### ISSUE TITLE
[python][dask] Reconnection warnings during DaskXGBClassifier fit

### ISSUE THREAD

**Issue by SimoneMatteo** [dmlc/xgboost#8227]
I'm trying to perform a DaskXGBClassifier fitting through a Dask cluster using a Dask Scheduler running in a Docker container, deployed in a Docker Swarm stack with 2 network interfaces. 

One of these interfaces has IP of 11.0.1.124 which [traefik](https://github.com/traefik/traefik) (a reverse proxy) is listening. Traefik is in place, in order to expose the Dask dashboard in a sensible url, so we can follow up on the internals of dask scheduling.  The dask scheduler is also listening on another subnet where all the other services are listening to, including the dask workers. 

The model fitting keeps getting a reconnection warning, but doesn't fail. The usual estimator logs are not displayed, but the fitting returns a model with low accuracy. 

However, there are also situations in which the fitting fails completely. 

 

Removing the traefik network subnet from our Docker Swarm stack solves the problem, but it isn't convenient because it prevents the Dask dashboard reachability. 

 

How should this be set up? Is there any way I can specify a subnet for the proper (internal, and not public) network interface to be correctly used?

Minimal code example: 
```python
import xgboost as xgb 

import dask.array as da 

 

from dask.distributed import Client 

client = Client("tcp://dask_dask_scheduler:8786") 

 

X_train = da.from_zarr("X_train2020.zarr") 

y_train = da.from_zarr("y_train2020.zarr") 

 

X_test = da.from_zarr("X_test2020.zarr") 

y_test = da.from_zarr("y_test2020.zarr") 



model = xgb.dask.DaskXGBClassifier( 

            n_estimators=100, 

            max_leaves=30,  

            objective="multiclass", 

            early_stopping_rounds=1, 

            metric="multi_logloss", 

            grow_policy="lossguide", 

            verbosity=0, 

            validate_parameters=True, 

            random_state=42) 

 

model.fit(X_train, y_train, eval_set=[(X_test, y_test)], verbose=10)
 ```

Logs from Dask workers: 

```
[09:28:52] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 1): [11.0.1.124] 

[09:16:55] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 1): [11.0.1.124] 

[09:19:08] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 2): [11.0.1.124] 

[09:21:23] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 3): [11.0.1.124] 

[09:23:38] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 4): [11.0.1.124] 

[09:25:56] WARNING: ../rabit/src/allreduce_base.cc:237: Connect to (failed): [11.0.1.124] 

[09:25:56] WARNING: ../rabit/src/allreduce_base.cc:446: failed in ReconnectLink [09:25:56] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out[09:25:56] WARNING: ../rabit/src/allreduce_base.cc:446: failed in ReconnectLink [09:25:56] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out[09:25:56] WARNING: ../rabit/src/allreduce_base.cc:446: failed in ReconnectLink [09:25:56] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out[09:25:56] WARNING: ../rabit/src/allreduce_base.cc:446: failed in ReconnectLink [09:25:56] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out 

[09:28:40] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 1): [11.0.1.124] 

[09:33:20] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 3): [11.0.1.124] 

[09:31:05] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 2): [11.0.1.124] 

[09:33:08] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 3): [11.0.1.124] 

[09:30:53] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 2): [11.0.1.124] 

[09:35:23] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 4): [11.0.1.124] 

[09:35:35] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 4): [11.0.1.124] 

[09:37:52] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 1): [11.0.1.124] 

[09:37:40] WARNING: ../rabit/src/allreduce_base.cc:139: Failed to shutdown due to[09:37:40] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out 

[09:37:40] WARNING: ../rabit/src/allreduce_base.cc:237: Connect to (failed): [11.0.1.124] 

[09:37:52] WARNING: ../rabit/src/allreduce_base.cc:237: Connect to (failed): [11.0.1.124] 

[09:37:52] WARNING: ../rabit/src/allreduce_base.cc:139: Failed to shutdown due to[09:37:52] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out 

[09:37:57] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 1): [11.0.1.124] 

[09:40:06] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 2): [11.0.1.124] 

[09:42:21] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 3): [11.0.1.124] 

[09:42:25] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 3): [11.0.1.124] 

[09:40:10] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 2): [11.0.1.124] 

[09:44:36] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 4): [11.0.1.124] 

[09:44:40] WARNING: ../rabit/src/allreduce_base.cc:240: Retry connect to ip(retry time 4): [11.0.1.124] 

[09:46:53] WARNING: ../rabit/src/allreduce_base.cc:237: Connect to (failed): [11.0.1.124] 

[09:46:53] WARNING: ../rabit/src/allreduce_base.cc:139: Failed to shutdown due to[09:46:53] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out 

[09:46:57] WARNING: ../rabit/src/allreduce_base.cc:237: Connect to (failed): [11.0.1.124] 

[09:46:57] WARNING: ../rabit/src/allreduce_base.cc:139: Failed to shutdown due to[09:46:57] ../rabit/include/rabit/internal/utils.h:108: Socket Connect Error:Connection timed out 
```

**Comment by trivialfis** [comment]
Hi, co you please visit https://xgboost.readthedocs.io/en/stable/tutorials/dask.html#troubleshooting and see if it can help?

**Comment by SimoneMatteo** [comment]
Thanks for the pointer. However, even if we set the scheduler_address through the dask.config setting, the warnings from the workers still complain about the wrong IP (no idea where they pick up that IP).

From the moment, we removed the double subnet from the Dask Scheduler, since this seems to solve our problem.

**Comment by trivialfis** [comment]
Could you please share the log from the scheduler as well?

**Comment by trivialfis** [comment]
Closing due to stalled.
